### PR TITLE
Change IV::cleanUpMinorVersions() and Throttle::purge() to use individual deletes instead of ranges

### PR DESCRIPTION
**Pull request by mreishus**
This should be better for the default "REPEATABLE READ" transaction level in InnoDB

### CODE REVIEW

File: Model/Throttle.php [hunk 1]
```diff
@@ -123,8 +123,13 @@ public function check($key, $allowed = 1) {
 		if (empty($allowed)) {
 			$allowed = 1;
 		}
-		$this->purge();
-		$count = $this->find('count', array( 'conditions' => compact('key') ));
+		register_shutdown_function(function() {
+			$this->purge();
```
Comment (zeroasterisk) [inline Model/Throttle.php]: does this work?  I thought you had to pass `use $this` to be able to use the `$this` object inside the callback function.

Comment (mreishus) [inline Model/Throttle.php]: Starting w/ 5.4 you can use $this in anonymous functions: http://php.net/manual/en/migration54.new-features.php

I've tested this through a controller calling ->throttle and it worked properly

**Comment by zeroasterisk** [comment]
:+1:

File: Model/Throttle.php (hunk 1 above)
Comment (zeroasterisk) [inline Model/Throttle.php]: Great - thanks!

**Comment by mreishus** [comment]
Fixed unit tests

**Pull request merged**